### PR TITLE
[Tizen] Updating the Tizen Client Generation Code

### PR DESCRIPTION
**Pull request by akhilkedia**
Reference Issue - #3076 

- Updated Tizen Client Codegen to remove calls to obsolete/deprecated system APIs.
- Auto-generation of the documentation of the produced client library using Doxygen.
- Included test cases for Petstore in /samples/client/petstore/tizen/PetStoreTest/

CC @maneeshsahu

Please feel free to contact me for any changes/corrections etc.
Signed-off-by: akhilkedia94 <akhil.kedia@samsung.com>

### CODE REVIEW

**Comment by wing328** [comment]
@akhilkedia thanks for the PR and including test cases to cover the change.

For Java file (e.g. TizenClientCodegen.java), please indent with 4-space instead of tab.

File: modules/swagger-codegen/src/main/resources/tizen/api-header.mustache [hunk 1]
```diff
@@ -1,37 +1,74 @@
-#ifndef {{classname}}_H_
-#define {{classname}}_H_
+/*
+ * Copyright (c) 2016 Samsung Electronics Co., Ltd All Rights Reserved
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/tizen/api-header.mustache]: @akhilkedia is it a must to include this particular line? For template files contributed by other companies or individuals, none of those includes the copyright note.

**Comment by akhilkedia** [comment]
@wing328 I have changed the indent, as well as removed the copyright notices.
Do tell me if any more changes etc. are needed.

**Comment by wing328** [comment]
@akhilkedia the change looks good to me. Can you please file the PR against `2.3.0` branch instead of `master` as I believe this change is breaking one (non-backward compatible)?

**Comment by akhilkedia** [comment]
@wing328 - I will file a new PR on the required branch.

**Pull request closed without merging**